brave/src/main/java/brave/CurrentSpanCustomizer.java

```diff
@@ -0,0 +1,56 @@
+package brave;
+
+/**
+ * Provides a mechanism for end users to be able to customise the current span.
+ *
+ * <p>Handles the case of there being no current span in scope.
+ */
+public final class CurrentSpanCustomizer implements SpanCustomizer {
```

Comment:
Thanks @adriancole! Nice way to deal with spans without bringing the whole tracer instance, the similar approach we are using in JAX-RS using contextual injection.